### DISCUSSION TITLE
FPGA Configuration from PS

### DISCUSSION THREAD

**Discussion by muratgureken** [polarfire-soc/polarfire-soc-documentation#350] [HSS and U-Boot]
Hello,
We are interested in use of Polarfire FPGA’s. I made research from your website and found some documents about the secure boot mode of Polarfire.
What I would like to learn is, is it possible to program PL (fpga) from PS in Polarfire? I think, as a flash-based FPGA, polarfire FPGA reads its configuration from its own flash. We would like to change and program the configuration of FPGA from PS, is it possible? If it is, how can we manage it?

Is it possible to manage/change FPGA configuration and boot settings form HSS?

Thanks,
Murat

**Comment by nitindeshpande** [comment]
Hello @muratgureken,
Thanks for your interest in PolarFire FPGAs.
There are two different devices [PolarFire FPGA](https://www.microchip.com/en-us/products/fpgas-and-plds/fpgas/polarfire-fpgas/polarfire-mid-range-fpgas) and [PolarFire SoC FPGA](https://www.microchip.com/en-us/products/fpgas-and-plds/system-on-chip-fpgas/polarfire-soc-fpgas).

I don't understand your requirement completely, but it is possible to re-program the FPGA part of the PolarFire SoC. The software running on PFSoC corecomplex helps in achieving that.

`On PolarFire SoC, auto update and In-Application Programming (IAP) are the mechanisms through which the FPGA can reprogram itself. In auto update, the device reprograms itself on power-up, and in IAP, the device is programmed when an application executing on the device requests it. In both cases, the system controller fetches a bitstream from external SPI flash & uses that to program the device.`

I would request you to refer to this document [re-programming-the-fpga-from-linux.md](https://github.com/polarfire-soc/polarfire-soc-documentation/blob/master/how-to/re-programming-the-fpga-from-linux.md) to see if that is what you are looking for.

The same feature is also available on PolarFire FPGA device as well. See [here](https://ww1.microchip.com/downloads/aemDocuments/documents/FPGA/ApplicationNotes/ApplicationNotes/microsemi_polarfire_fpga_auto_update_and_in_application_programming_application_note_ac466_v7.pdf).
